### PR TITLE
Makes the radiation gene better

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
- The radiation gene now makes its user shed a serious amount of radiation at a time, does not make them immune to overwhelming amounts of radiation.
- People with the radiation gene can wear clothing such as radiation suits to prevent irradiating everyone near them, but this also prevents emitting harvestable radiation.
- Massively improved the amount of harvestable radiation (the radiation that Collectors can use to generate power, the sort that are near Singularity engines) generated, it now depends on the user's total amount of radiation before shedding some of it, which means you can now capitalize on it for possibly massive power gains. Coincidentally I must inform people that the Particle Accelerator irradiates people.
- Increased range of harvestable radiation by 1.

No longer just an "i griff and die" gene, now this has actual uses for a disability.
:cl:
 * rscadd: The radiation gene now makes people shed off a massive amount of radiation instead of passively being irradiated.
 * tweak: Radiation emitted from the radiation gene can now be prevented by having the users wear hazmat suits. No longer involuntarily kill a lot of people by just standing near them! But at the same time it prevents users from emitting radiation that can be harvested.
 * tweak: Massively improved the amount of radiation from the radiation gene that can be harvested by radiation collectors, it now depends on how irradiated the user is. Yes, you can likely capitalize on this power production method. Wink wink, nudge nudge.
 * tweak: The range of radiation that can be harvested from the radiation gene has been increased by 1 tile.
 * tweak: Changed the activation and deactivation messages of the radiation gene.

### CODE REVIEW

File: code/game/dna/genes/goon_disabilities.dm [hunk 1]
```diff
@@ -27,24 +27,25 @@
...
-
-	emitted_harvestable_radiation(get_turf(owner), 1, range = 2) //Around 70W, nothing much really
+/datum/dna/gene/disability/radioactive/OnMobLife(var/mob/living/owner)
+	var/radiation = owner.radiation
+	owner.radiation = max(radiation - 30, 0)
+	if(owner.getarmor(null, "rad") < 100)
+		emitted_harvestable_radiation(get_turf(owner), radiation * 100, range = 3) //1 power = ~70W, are you ready for radiation engines?
+		for(var/mob/living/L in range(1, owner))
+			if(L == owner)
+				continue
```
Comment (Exxion) [inline code/game/dna/genes/goon_disabilities.dm]: Just use `orange()`

**Comment by GreatBigFailure** [comment]
Neat, I like the synergy it has with other rad engines. Makes geneticists contribute more than the eat gene

**Comment by Eneocho** [comment]
This will heavily affect Ghoulification, since the rad-shedding you get from the gene (which has an almost guaranteed chance to be added after reaching heavy radpoisoning) will directly counteract the radpumping. How often is the radpump? Once every tick?

**Comment by SonixApache** [comment]
suggest you condense the :cl: to something the average player could read without losing interest

**Comment by Eneocho** [comment]
Funny, I've heard the exact opposite opinion from players. That the changelogs are too short or obtuse.

**Comment by boy2mantwicethefam** [comment]
@Eneocho Once every tick. Ghoulification can be done without having the gene active, just irradiate someone in a conventional way
@SonixApache Are there players that are impatient to read the changelog? This would be new to me, because I always spend the time reading changelogs before round-start since there's nothing else to do but wait for the round to begin

**Comment by Eneocho** [comment]
>Ghoulification can be done without having the gene active, just irradiate someone in a conventional way

That ain't how it works, you WILL get the radioactive gene. The odds are dependent on how many rads you have, which eventually leads to a 100% chance for it to activate. Irradiating in a "conventional" way will still activate it regardless.

**Comment by SonixApache** [comment]
ok barry, now code a cell that stores no power but runs off radiation so that people with this gene can power their own hardsuits

**Comment by boy2mantwicethefam** [comment]
@Eneocho I can't find the relevant code that determines where mutation from radiation happens, all I can find is code that has been commented out
Interestingly enough if you manage to keep someone at a steady low dose you can avoid most negative radiation effects until 20 minutes (600 ticks) have passed and then you can bathe them into a huge amount of radiation to turn them into ghouls, but the chance is still very low (500 radiation has a 0.5% chance to trigger ghoulification)
@SonixApache Hardsuits that offer radiation protection block the user from emitting radiation, but I COULD make a different PR where it's powered by radiation and offers a few seconds of power every time it grabs harvestable radiation

**Comment by Eneocho** [comment]
There's M_GLOW on the handle for mutations and radiation file. I think it's a different gene than radblock But not sure if it pulls code from the radiation gene.

**Comment by Blithering** [comment]
You could also keep SE injectors handy to deactivate the rad gene in case it activates.

**Comment by Eneocho** [comment]
> You could also keep SE injectors handy to deactivate the rad gene in case it activates.

That doesn't work, since ghoulification also gives you the M_NOCLONE flag, which stops gene editing and cloning.

**Comment by boy2mantwicethefam** [comment]
I can improve the chances of ghoulification in a different PR

**Comment by Eneocho** [comment]
>I can improve the chances of ghoulification in a different PR

No need. It was mostly a small fear due to ghoulification already being a very obscure and hard thing to achieve.
I've been meaning to remove the NOCLONE flag upon turning into ghoul but haven't done it yet.

**Pull request merged**